Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/self_intersections.h

```diff
@@ -503,11 +504,21 @@ bool does_self_intersect(const FaceRange& face_range,
     CGAL::Emptyset_iterator unused_out;
     internal::self_intersections_impl<ConcurrencyTag>(face_range, tmesh, unused_out, true /*throw*/, np);
   }
-  catch(CGAL::internal::Throw_at_output_exception&)
+  catch (const CGAL::internal::Throw_at_output_exception&)
   {
     return true;
   }
-
+  #if defined(CGAL_LINKED_WITH_TBB)
```

Comment:
My suggestion to remove the deprecation warning: only use that code when necessary (on MacOS with odd TBB versions):
```suggestion
  #if defined(CGAL_LINKED_WITH_TBB) && TBB_USE_CAPTURED_EXCEPTION
```

Reply:
In fact, they deprecated the whole tbb exception concept so it all will be gone in the next versions. Maybe it is already in version 2021 but brew provides only the formula for the 2020 version.

Reply:
I will check your suggestion. It sounds good.